### ISSUE TITLE
Adapting nvQuickTheme for other CSS frameworks other than Bootstrap

### ISSUE THREAD

**Issue by skrantzman** [nvisionative/nvQuickTheme#226]
## Please summarize your question in one sentence
Can the Bootstrap portion of nvQuickTheme be swapped out for another CSS framework?

## Steps to reproduce
I was thinking all we would have to do is remove the Bootstrap dependencies in the package.json file and replace them with the dependencies for a different CSS framework.  Then update the includes.ascx file with the new framework includes, and replace all of the Bootstrap classes for the new framework classes in the remaining .ascx files.  

Is that all it would take, or is there more to it than that?

## Other comments or remarks
Although I have never designed a DNN skin in Tailwind CSS, I have used it in other projects and find it quite useful. It might be nice to have a version of nvQuickTheme for Tailwind CSS.

**Comment by david-poindexter** [comment]
@skrantzman thanks for this question.  We get it from time to time, so it is nice to document the answer here.  In short though, yes - that's going down the right track.  In addition to what you have stated, you would also need to update gulpfile.js accordingly to accommodate whatever tasks are appropriate for that particular CSS framework.  We have done this for "no framework" solutions, Foundation, and a few Material based CSS frameworks.  It works great and once you have a new base, you can just maintain that base as a starting point moving forward for similar projects.

**Comment by skrantzman** [comment]
@david-poindexter, thank you for the quick response.  if I manage to get it working for Tailwind CSS, would it be appropriate to post a public gist of it on githuyb for other to use?

**Comment by ahoefling** [comment]
@skrantzman I am a big Foundation fan and prefer it to many sites over Bootstrap. It is possible to use nvQuickTheme with other libraries. Echoing what @david-poindexter said you will need to make significant updates to the gulpfile.js as well as all the sass, js, and ascx files. It is quite a bit of work but once done it works just like it does with bootstrap!

Good Luck 💯

**Comment by skrantzman** [comment]
@ahoefling, Thanks for the encouragement.  You are right about the time, as there is more involved in porting it over to Tailwind than I originally thought. I am about 3/4 of the way there, I just have to implement purging of Tailwind's unused css, fine tune the default theme and run some debugging. I am glad I took the project on though, as I have learned quite a bit more about nvQuickTheme, TailwindCss, and Gulp in the process. [DNN_Tailwind](https://github.com/skrantzman/DNN_Tailwind)
